roles/unprivileged_user/tasks/main.yml

```diff
@@ -66,3 +66,40 @@
     - not authorized_keys_file.stat.exists
     - unprivileged_user_import_ssh_pub_key | default(True)
   become: true
+
+- name: "Configure {{ unprivileged_user_username }} for Kerberos authentication"
+  block:
+    - name: "Install krb5-workstation on Red Hat based distributions"
+      ansible.builtin.dnf:
+        name:
+          - "krb5-workstation"
+          - "krb5-libs"
+        state: present
+      when: ansible_os_family == "RedHat"
+
+    - name: "Install krb5-user on Debian based distributions"
+      ansible.builtin.apt:
+        name: "krb5-user"
+        state: present
+      when: ansible_os_family == "Debian"
+
+    - name: "Copy Kerberos client config from Host"
```

Comment:
This seriously feels out of scope for this role. Roles should be small and focused. When I read `unprivileged_user` I don't think it essentially means "reconfigure the whole system".

What I'd do is create a role to make the system a kerberos client, with an option to copy from the host. You can then combine those in a playbook. Composition is a great thing and playbooks are the right place for that.

Reply:
With the default parameter values, the behavior is exactly the same because none of the extra tasks here will run. But I don't mind extracting it out into a separate role to keep things organized.